src/renderer/services/file-management-system/index.ts

```diff
@@ -398,85 +397,134 @@ export default class FileManagementSystem {
     const fileLastModifiedInMs = fileLastModified.getTime();
-
     if (lastModifiedInMS !== fileLastModifiedInMs) {
       throw new Error("File has been modified since last upload attempt");
     }
-
-    // Skip trying to resume if there is no FSS upload to check
-    if (fssUploadId) {
-      // Retrive job FSS uses to track its upload portion
-      let fssUpload;
-      try {
-        fssUpload = (await this.jss.getJob(fssUploadId)) as FSSUpload;
-      } catch (error) {
-        // Because FSS uses a queue to interact with JSS this job
-        // may not exist even though the upload is in progress
-        // in which case the app can just wait it out
-        console.error("Failed to get FSS job")
-        return;
-      }
-
-      if (fssStatus.status === UploadStatus.COMPLETE) {
+    switch (fssStatus.status) {
+      case UploadStatus.WORKING:
+        // Update status to reflect the resume going smoothly
+        await this.jss.updateJob(fuaUpload.jobId, {
+          status: JSSJobStatus.RETRYING,
+        });
+        if (localNasShortcut) {
+          // For localNasShortcut uploads, the way to reume an in progress upload is to call /register on it again. 
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const [registration, _s, fileSize] = await this.register(fuaUpload);
+          this.updateProgress(registration.uploadId, fileSize, (progress: UploadProgressInfo) => onProgress(registration.uploadId, progress))
+        } else {
+          await this.resumeUploadInChunks(fuaUpload, fssStatus, onProgress);
+        }
+        break;
+      case UploadStatus.RETRY:
+        if (localNasShortcut) {
+          await this.fss.retryFinalizeLocalNasShortcut(fssStatus.uploadId);
+        } else {
+          await this.retryChunkedUpload(fssStatus);
+        }
+        break;
+      case UploadStatus.COMPLETE:
         // If an FSS upload status is complete it has performed everything
         // it needs to and may just need the client to finish its portion
-        const { fileId } = fssUpload.serviceFields;
+        const { fileId } = fssStatus;
         if(!fileId){
           throw new Error("FileId was not published on COMPLETE upload: " + fssStatus.uploadId)
         }
-        await this.complete(upload, fileId)
-      } else if (fssStatus.status === UploadStatus.RETRY || fssStatus.status === UploadStatus.WORKING) {
-        // Update status to reflect the resume going smoothly
-        await this.jss.updateJob(upload.jobId, {
-          status: JSSJobStatus.RETRYING,
-        });
+        await this.complete(fuaUpload, fileId)
+        break;
+      case UploadStatus.POST_PROCESSING:
+        onProgress(fssStatus.uploadId, { bytesUploaded: fssStatus.currentFileSize, totalBytes: fssStatus.currentFileSize });
+        break;
+      default:
+        console.error(`Unexpected FSS UploadStatus encountered: ${fssStatus?.status}`);
```

Comment:
I agree about throwing an error, but if this is reached, that means they are using a client version that is out of date with the FSS2 API.  They should be prompted to update to the latest version, right?

Reply:
>They should be prompted to update to the latest version, right?

Yes though not because of this error, if they have gotten here they have already ignored that prompt. Regardless, I think throwing an error here is more visible

Reply:
Converted to an error.

About the version prompt, i was responded to your idea that they should be prompted to retry.  If their version is out of date, then retrying again is probably not advisable until they update the version.